configure.ac

```diff
@@ -897,6 +897,32 @@ else
   GRN_WITH_LZ4=no
 fi
 
+# LuaJIT
+AC_ARG_WITH(luajit,
+  [AS_HELP_STRING([--with-luajit],
+    [Support LuaJIT. [default=auto]])],
+  [with_luajit="$withval"],
+  [with_luajit="auto"])
+if test "x$with_luajit" != "xno"; then
+  m4_ifdef([PKG_CHECK_MODULES], [
+    PKG_CHECK_MODULES([LUAJIT],
+                      [luajit],
+                      [GRN_WITH_LUAJIT=yes],
+                      [GRN_WITH_LUAJIT=no])
+  ],
+  [GRN_WITH_LUAJIT=no])
+  if test "$GRN_WITH_LUAJIT" = "yes"; then
+    AC_DEFINE(GRN_WITH_LUAJIT, [1],
+              [Support LuaJIT.])
+  else
+    if test "x$with_luajit" != "xauto"; then
+      AC_MSG_ERROR("No libluajit found")
+    fi
+  fi
+else
+  GRN_WITH_LUAJIT=no
+fi
```

Comment:
We need `AC_SUBST(LUAJIT_CFLAGS)` and `AC_SUBST(LUAJIT_LIBS)` to use them in `Makefile.am`.